cmake/partition_manager.cmake

```diff
@@ -4,6 +4,60 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+# set_dfu_hex_offset([EXT_START <address> EXT_BASE <address>])
+function(set_dfu_hex_offset)
+  cmake_parse_arguments(SEC_SLOT "" "EXT_START;EXT_BASE" "" ${ARGN})
+
+  # Check if a signed version of the network core application is defined.
+  # If so, this indicates that we need to support firmware updates on the
+  # network core. This again means that we should generate the required
+  # hex files.
+  get_shared(cpunet_signed_app_hex IMAGE CPUNET PROPERTY PM_SIGNED_APP_HEX)
+
+  if (DEFINED SEC_SLOT_EXT_START)
+    if (NOT DEFINED SEC_SLOT_EXT_BASE)
+      message(FATAL_ERROR "EXT_BASE must be set when EXT_START is set")
+    endif()
+    set(ext_offset "${SEC_SLOT_EXT_START} + ${SEC_SLOT_EXT_BASE} +")
+  endif()
+
+  if (CONFIG_NRF53_UPGRADE_NETWORK_CORE
+      AND DEFINED cpunet_signed_app_hex)
+    # The address coming from other domains are not available in this scope
+    # since it is imported by a different domain. Hence, it must be fetched
+    # through the 'partition_manager' target.
+    get_target_property(net_app_addr partition_manager CPUNET_PM_APP_ADDRESS)
+
+    # Calculate the offset fram the address which the net/app core app is linked
+    # against to the secondary slot. We need these values to generate hex files
+    # which targets the secondary slot.
+    set(net_to_sec_sum
+      "${ext_offset} ${PM_MCUBOOT_SECONDARY_ADDRESS} - ${net_app_addr} + ${PM_MCUBOOT_PAD_SIZE}")
+    math(EXPR net_app_to_secondary ${net_to_sec_sum})
```

Comment:
I find this calculation very hard to read, especially the missing operator between: `${ext_offset} ${PM_MCUBOOT_SECONDARY_ADDRESS}`.
Why not do like this:
```suggestion
      
    math(EXPR net_app_to_secondary
      "${SEC_SLOT_EXT_START} \
       + ${SEC_SLOT_EXT_BASE} \
       + ${PM_MCUBOOT_SECONDARY_ADDRESS} \
       - ${net_app_addr} \
       + ${PM_MCUBOOT_PAD_SIZE}"
    )
```

CMake allows math code like this:
```
math(EXPR result "+ + 5 + 12 + 20")
message("result=${result}")
```
gives:
```
result=37
```
so no need to worry about extra `+` operators.

Reply:
Wow that was much better, thanks!